var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -224,6 +225,11 @@ def install(self, spec, prefix):
         prefix_path = prefix.bin if '@:5.4.0' in spec else prefix
         options = ['-prefix={0}'.format(prefix_path)]
 
+        # Needed anytime target architecture flags leads to a binary
+        # that cannot be executed on the host architecture
+        if '+crosscompile' in spec:
+            options.append('--host')
```

Comment:
Isn't it possible to avoid the variant completely and just compare `host` and `target` of the spec?

Reply:
Yes, I would like to avoid the additional variant as well. I am assuming that the default behavior is that the `host` and `target` should match unless the Spack config files say otherwise? Unless there is some cross-compile boolean set somewhere, I will have to do a string comparison.

Reply:
You can detect the host using:
```python
import llnl.util.cpu

host = llnl.util.cpu.host()
```
and then compare `host.family` to `self.spec.target.family` to check whether you are cross-compiling or not. Would that solve the issue here?